app/src/main/java/com/team/uptech/pomodoro/data/repository/impl/PomodoroRepositoryPrefsImpl.kt

```diff
@@ -0,0 +1,91 @@
+package com.team.uptech.pomodoro.data.repository.impl
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import android.util.Log
+import com.team.uptech.pomodoro.data.model.Pomodoro
+import com.team.uptech.pomodoro.data.repository.PomodoroRepository
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+/**
+ * Created on 28.04.17.
+ */
+class PomodoroRepositoryPrefsImpl @Inject constructor(context: Context) : PomodoroRepository {
+
+    private val prefs: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+
+    override fun getCurrentPomodoro(): Single<Pomodoro> {
+        return Single.create<Pomodoro> { sb ->
+            val pomodoroTypeString = getPomodoroType()
+            val pomodoro = Pomodoro.valueOf(pomodoroTypeString)
+            getPomodoroTypeTime(pomodoro)
+                    .subscribe({
```

Comment:
a subscribe() inside another Observable creation is usually a bad decision that can be rewritten with a flatMap